src/test/java/plugins/JobDslPluginTest.java

```diff
@@ -470,10 +472,47 @@ public void should_fail_on_missing_plugin() {
                          "}");
         seedJob.save();
         Build build = seedJob.scheduleBuild().shouldBeUnstable();
+        assertThat(build.getConsole(), containsString("plugin 'chucknorris' needs to be installed"));
+
+        seedJob.configure(() -> jobDsl.setFailOnMissingPlugin(true));
+        build = seedJob.scheduleBuild().shouldFail();
+        assertThat(build.getConsole(), containsString("plugin 'chucknorris' needs to be installed"));
+    }
+
+    /**
+     * Verifies whether the build will be marked as failed or unstable if a plugin must
+     * be installed to support all features used in the DSL script. To check the right behavior we used the old assert
+     * which is failing on Java 11. This test can be used to fix the root cause.
+     */
+    @Ignore("Currently failing on Java 11, use this test to fix https://issues.jenkins-ci.org/browse/JENKINS-54942")
+    @Issue("https://issues.jenkins-ci.org/browse/JENKINS-54942")
+    @Test
+    public void should_fail_on_missing_plugin_but_doesnt_on_java_11() {
```

Comment:
The old test renamed and ignored to help when addressing the bug

Reply:
Frankly, I do not feel strongly. You are right the part that causes assertion to fail is sort of cosmetic and the main thing is correct here (the missing plugin is repported correctly). I am ok narrowing the scope of the test to move forward here. On the other hand, we caught a regression thanks to that and it makes sense to keep it to work as a regression test for plugin authors.

Reply:
Are you suggesting to keep this test working, remove the @Ignore? Or just close the PR and keep the test failing? What I wouldn't like to loose is the fact that the test is doing the job, although without the information regarding the line. Perhaps just a comment on the original test warning about that to help when we address the bug itself. WDYT?